models/models.go

```diff
@@ -73,56 +58,53 @@ type QueryParams struct {
 	Resource    string
 }
 
-// TODO: Seperate database struct and request struct
 type ExportPayload struct {
-	ID          uuid.UUID      `gorm:"type:uuid;primarykey" json:"id"`
-	CreatedAt   time.Time      `gorm:"autoCreateTime" json:"created"`
-	UpdatedAt   time.Time      `gorm:"autoUpdateTime" json:"-"`
-	CompletedAt *time.Time     `json:"completed,omitempty"`
-	Expires     *time.Time     `json:"expires,omitempty"`
-	RequestID   string         `json:"-"`
-	Name        string         `json:"name"`
-	Format      PayloadFormat  `gorm:"type:string" json:"format"`
-	Status      PayloadStatus  `gorm:"type:string" json:"status"`
-	Sources     datatypes.JSON `gorm:"type:json" json:"sources"`
-	S3Key       string         `json:"-"`
+	ID          uuid.UUID `gorm:"type:uuid;primarykey"`
+	CreatedAt   time.Time `gorm:"autoCreateTime"`
+	UpdatedAt   time.Time `gorm:"autoUpdateTime"`
+	CompletedAt *time.Time
+	Expires     *time.Time
+	RequestID   string
+	Name        string
+	Format      PayloadFormat `gorm:"type:string"`
+	Status      PayloadStatus `gorm:"type:string"`
+	Sources     []Source      `gorm:"foreignKey:ExportPayloadID"`
+	S3Key       string
 	User
 }
 
 type Source struct {
-	ID          uuid.UUID      `json:"id"`
-	Application string         `json:"application"`
-	Status      ResourceStatus `json:"status"`
-	Resource    string         `json:"resource"`
-	Filters     datatypes.JSON `json:"filters"`
+	ID              uuid.UUID `gorm:"type:uuid;primarykey"`
+	ExportPayloadID uuid.UUID `gorm:"type:uuid"`
+	Application     string
+	Status          ResourceStatus
+	Resource        string
+	Filters         datatypes.JSON `gorm:"type:json"`
 	*SourceError
 }
 
 type SourceError struct {
-	Message string `json:"message"`
-	Code    int    `json:"code"`
+	Message string
+	Code    int
 }
 
 type User struct {
-	AccountID      string `json:"-"`
-	OrganizationID string `json:"-"`
-	Username       string `json:"-"`
+	AccountID      string
+	OrganizationID string
+	Username       string
 }
 
-func (ep *ExportPayload) BeforeCreate(tx *gorm.DB) error {
+func (ep *ExportPayload) BeforeCreate(tx *gorm.DB) (err error) {
 	ep.ID = uuid.New()
-	ep.Status = Pending
-	sources, err := ep.GetSources()
-	if err != nil {
-		return err
+	if ep.Expires == nil {
+		expirationTime := time.Now().AddDate(0, 0, config.ExportCfg.ExportExpiryDays)
+		ep.Expires = &expirationTime
 	}
```

Comment:
Does the service need to accept the `expires` date?  Or is this a field that should be filled in by the service?

Reply:
If the request doesn't specify expires, then the service fills it in with the default.

Reply:
In that case, does the `ExportRequest` in the openapi spec need to be modified to include an optional `expires` field?